src/main/kotlin/com/exactpro/th2/codec/AbstractCodecProcessor.kt

```diff
@@ -80,13 +80,9 @@ abstract class AbstractCodecProcessor(
         }
     }
 
-    protected fun Set<String>.reportWarnings(context: ReportingContext, action: String, messagesIds: () -> List<MessageID> = { emptyList() }) {
-        val warnings = context.warnings
-        if (warnings.isNotEmpty()) {
-            when (warnings.size) {
-                1 -> onEvent("[WARNING] ${warnings.single()} during $action", messagesIds())
-                else -> onEvent("[WARNING] Message $action produced ${warnings.size} warnings", messagesIds(), body = warnings)
-            }
+    protected fun Set<String>.onEachWarning(context: ReportingContext, action: String, messagesIds: () -> List<MessageID> = { emptyList() }) = context.warnings.let { warnings ->
+        warnings.forEach { warning ->
+            this.onEachEvent("[WARNING] Message $action produced ${warnings.size} warnings", messagesIds(), body = listOf(warning))
```

Comment:
Why do we need to report a number of warnings in an event that contain only a single warning? I think there is no needs for that and we can omit that part in event's name